bucoffea/plot/studies/stack_plot/distributions.py

```diff
@@ -4,78 +4,25 @@
-    'ak4_phi1' : Bin("jetphi", r"Trailing AK4 jet $\phi$", 50,-np.pi, np.pi),
-    'ak4_eta0' : Bin("jeteta", r"Leading Jet $\eta$", 50, -5, 5),
-    'ak4_eta1' : Bin("jeteta", r"Leading Jet $\eta$", 50, -5, 5),
-    'ak4_central_eta' : Bin("jeteta", r"More Central Jet $\eta$", 50, -5, 5),
-    'ak4_forward_eta' : Bin("jeteta", r"More Forward Jet $\eta$", 50, -5, 5),
-    'ak4_nef0' : Bin('frac', 'Leading Jet Neutral EM Frac', 50, 0, 1),
-    'ak4_nef1' : Bin('frac', 'Trailing Jet Neutral EM Frac', 50, 0, 1),
-    'ak4_nhf0' : Bin('frac', 'Leading Jet Neutral Hadronic Frac', 50, 0, 1),
-    'ak4_nhf1' : Bin('frac', 'Trailing Jet Neutral Hadronic Frac', 50, 0, 1),
-    'ak4_chf0' : Bin('frac', 'Leading Jet Charged Hadronic Frac', 50, 0, 1),
-    'ak4_chf1' : Bin('frac', 'Trailing Jet Charged Hadronic Frac', 50, 0, 1),
-    # 'dphitkpf' : Bin('dphi', r'$\Delta\phi_{TK,PF}$', 50, 0, 3.5),
-    'met' : Bin('met',r'$p_{T}^{miss}$ (GeV)',list(range(0,500,50)) + list(range(500,1000,100)) + list(range(1000,2000,250))),
-    'met_phi' : Bin("phi", r"$\phi_{MET}$", 50, -np.pi, np.pi),
-    'ak4_mult' : Bin("multiplicity", r"AK4 multiplicity", 10, -0.5, 9.5),
-    'ak4_mt0' : Bin("mt", r"Leading AK4 $M_{T}$ (GeV)", 50, 0, 1000),
-    'ak4_mt1' : Bin("mt", r"Trailing AK4 $M_{T}$ (GeV)", 50, 0, 1000),
-    'dphijr' : Bin("dphi", r"$min\Delta\phi(j,recoil)$", 50, 0, 3.5),
-    'extra_ak4_mult' : Bin("multiplicity", r"Additional AK4 Jet Multiplicity", 10, -0.5, 9.5),
-    'recoil' : Bin('recoil','Recoil (GeV)', recoil_bins_2016),
-}
-
-ylims = {
-    'ak4_eta0' : (1e-3,1e8),
-    'ak4_eta1' : (1e-3,1e8),
-    'ak4_nef0' : (1e0,1e8),
-    'ak4_nef1' : (1e0,1e8),
-    'ak4_nhf0' : (1e0,1e8),
-    'ak4_nhf1' : (1e0,1e8),
-    'ak4_chf0' : (1e0,1e8),
-    'ak4_chf1' : (1e0,1e8),
-    'vecb' : (1e-1,1e9),
-    'vecdphi' : (1e0,1e9),
-    'dphitkpf' : (1e0,1e9),
-    'met' : (1e-3,1e5),
-    'ak4_mult' : (1e-1,1e8),
-    'particlenet_score' : (1e-1,1e8),
-}
+    'sr_vbf'    :         common_distributions + obj_variables(object_name='ak4',       indices=[0,1], vars=['nef','nhf','chf']),
+    'sr_vbf_nodijetcut' : common_distributions + obj_variables(object_name='ak4',       indices=[0,1], vars=['nef','nhf','chf']),
+    'cr_1m_vbf' :         common_distributions + obj_variables(object_name='muon',      indices=[],    vars=['pt', 'eta', 'phi'], extravars=['mt']),
+    'cr_1e_vbf' :         common_distributions + obj_variables(object_name='electron',  indices=[],    vars=['pt', 'eta', 'phi'], extravars=['mt']),
+    'cr_2m_vbf' :         common_distributions + obj_variables(object_name='muon',      indices=[0,1], vars=['pt', 'eta', 'phi']) + ['dimuon_mass'],
+    'cr_2e_vbf' :         common_distributions + obj_variables(object_name='electron',  indices=[0,1], vars=['pt', 'eta', 'phi']) + ['dielectron_mass'],
+    'cr_g_vbf'  :         common_distributions + obj_variables(object_name='photon',    indices=[0],   vars=['pt', 'eta', 'phi']),
+}
```

Comment:
Thanks for this restructuring, I think it looks really cool and easier to understand.